robot/src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -27,28 +32,39 @@ public class Drive extends DreadbotSubsystem {
     private final MotorControllerGroup rightMotors;
 
     protected final SlewRateLimiter slewRate;
+    private final DifferentialDriveOdometry odometry;
 
-    public Drive() {
+    private final AHRS gryo;
+
+    public Drive(AHRS gryo) {
         this.frontLeftMotor = new DreadbotMotor(new CANSparkMax(MotorConstants.FRONT_LEFT_MOTOR_PORT, MotorType.kBrushless), "frontLeft");
         this.frontRightMotor = new DreadbotMotor(new CANSparkMax(MotorConstants.FRONT_RIGHT_MOTOR_PORT, MotorType.kBrushless), "frontRight");
         this.backLeftMotor = new DreadbotMotor(new CANSparkMax(MotorConstants.BACK_LEFT_MOTOR_PORT, MotorType.kBrushless), "backLeft");
         this.backRightMotor = new DreadbotMotor(new CANSparkMax(MotorConstants.BACK_RIGHT_MOTOR_PORT, MotorType.kBrushless), "backRight");
-        
+
         frontLeftMotor.setIdleMode(IdleMode.kBrake);
         frontRightMotor.setIdleMode(IdleMode.kBrake);
         backLeftMotor.setIdleMode(IdleMode.kBrake);
         backRightMotor.setIdleMode(IdleMode.kBrake);
 
-        frontLeftMotor.setInverted(true);
-        backLeftMotor.setInverted(true);
+        frontLeftMotor.setInverted(false);
         frontRightMotor.setInverted(false);
+        backLeftMotor.setInverted(false);
         backRightMotor.setInverted(false);
+        
 
         leftMotors = new MotorControllerGroup(frontLeftMotor.getSparkMax(), backLeftMotor.getSparkMax());
         rightMotors = new MotorControllerGroup(frontRightMotor.getSparkMax(), backRightMotor.getSparkMax());
 
+        leftMotors.setInverted(false);
+        rightMotors.setInverted(true);
         diffDrive = new DifferentialDrive(leftMotors, rightMotors);
-
+        this.gryo = gryo;
+        odometry = new DifferentialDriveOdometry(
+            gryo.getRotation2d(),
+            (frontLeftMotor.getEncoder().getPosition() / AutonomousConstants.ROTATIONS_PER_METER),
+            (frontRightMotor.getEncoder().getPosition() / AutonomousConstants.ROTATIONS_PER_METER)
```

Comment:
Here we use the individual motor encoder, and the value needs be correctly inverted

Setting the inverted on the motors would work here, setting it on the motorGroups would not.